frontend/src/pages/Profile.js

```diff
@@ -28,60 +111,245 @@ import { textColor } from "../assets/jss/colorPalette";
+			inputDescription: emailRequirementsText,
+		},
+		{
+			label: passwordLabel,
+			inputFieldList: [
+				{ label: "Old password", required: false, type: "password", },
+				{ label: "New password", required: false, type: "password", },
+				{ label: "Repeat new password", required: false, type: "password", },
+			],
+			inputColumnExtra: passwordChangeStatus,
+			inputDescription: passwordRequirementsText,
+		},
+		{
+			label: "Wallet status",
+			inputColumnExtra: <WalletStatus user={user} setUser={setUser} />,
+			inputDescription: "Connect your existing Metamask wallet to NFT The World account",
+		},
+		{
+			label: "Log out",
+			inputColumnExtra: <LogoutButton setUser={setUser} changeRoute={changeRoute} />, 
+		},
+		{
+			inputColumnExtra: saveButton,
+		}
+	];
+
+	// TODO live text input validation
+	const pictureLabel = "Userpic";
+	const pictureRequirementsText = "max 180x180";  // TODO
+	const usernameLabel = "Username";
+	const usernameRequirementsText = "letters, digits, punctuation";  // TODO
+	const emailLabel = "Email";
+	const emailRequirementsText = "letters, digits, .+-@ ";
+	const passwordLabel = "Change password";
+	const passwordRequirementsText = "";  // TODO, I find password limitations useless
+
+
+	const handleChangePassword = (event) => {
+		const data = new FormData(event.currentTarget);
+		const [oldPassword, newPassword, repeatedPassword] = [data.get(`${passwordLabel}0`), data.get(`${passwordLabel}1`), data.get(`${passwordLabel}2`)];
+
+		if (newPassword === repeatedPassword)
+			appwriteApi.changePassword(oldPassword, newPassword)
+				.then(() => {
+					setHasPasswordChanged(true);
+					setPasswordErrorText("");
+				})
+				.catch(err => setPasswordErrorText(err.message));
+		else
+			setPasswordErrorText("The repeated password does not coincide.");
+	};
+
+	const checkAndSaveProfile = (event) => {  // TODO
+		event.preventDefault();
+		handleChangePassword(event);
+	};
+
+
+	const isEmailVerified = user.emailVerification;
+
+	const emailVerificationStatus = <StatusMessage isSuccessful={isEmailVerified} successText="Email verified" otherText="Email not verified" />;
+	const hasTriedPasswordChange = hasPasswordChanged || !!passwordErrorText;
+	const passwordChangeStatus = hasTriedPasswordChange && <StatusMessage isSuccessful={hasPasswordChanged} successText="Password has changed" otherText={passwordErrorText} />;
+
+	const EmailStatusBanner = ({ isVerified }) => {
+		const emailStatusBannerStyle = {
+			paddingTop: "11px", paddingLeft: "20px", paddingBottom: "11px", paddingRight: "20px",
+			borderRadius: "11px",
+		};
+
+		return (isVerified)?
+			(<div style={{ background: "#D7FFDD", ...emailStatusBannerStyle, display: "flex", alignItems: "center", }}>
+				<GreenCheck/>
+
+				<Margin width="13px"/>
+
+				<ParagraphTypography style={{ display: "inline", fontWeight: "400", fontSize: "15px", color: "#456548", }}>
+					Your email was successfully confirmed
+				</ParagraphTypography>
```

Comment:
Shouldn't this be shown only once and not every time?

Reply:
I think we do not need to display the information at all if the mail is verified as this info is also displayed on the profile page anyways.